### PR TITLE
Add flag for UIWebView usage

### PR DESCRIPTION
**Pull request by bbialas**

### CODE REVIEW

**Comment by mihaimuntenas** [comment]
Microsoft people, Apple is killing us :) ... Does anyone care to take a look at this?

**Comment by ludufre** [comment]
@Microsoft merge this and publish on npm please. 🙃

**Comment by bbialas** [comment]
ping...

**Comment by ckarlss0n** [comment]
@ermik The README has been updated by the original author.

There are numerous issues posted about CodePush and WKWebView not being compatible – which is a very serious issue as WKWebView is the new standard. Are you working on making the two compatible with each other, and if so, when can we expect to see this fix?

We'd love to hear some updates regarding this, so we know if we should wait a while or try to find some alternative to CodePush instead.

Thanks!

**Comment by jacobg** [comment]
It's very sad that this pull request has been wasting away for months.

**Comment by brassier** [comment]
It's unfortunate that this is sitting idle.  Hopefully @ahdbilal will have insight by tomorrow [as promised](https://github.com/microsoft/cordova-plugin-code-push/issues/594#issuecomment-601801926).

Besides the fact that this isn't merged in, I have 2 questions:
1) Has anyone tried using this successfully to get a functioning codepush plugin that Apple doesn't reject?
2) Any idea if this will still work in `cordova-ios@6.x`?  Looks like there are some heavy changes to the use of `WK_WEB_VIEW_ONLY` at 6.x.

**Comment by l3ender** [comment]
Hello @ahdbilal, any update on this getting merged/release? It is critical because Apple will be rejecting app submissions unless this issue is resolved. See https://github.com/microsoft/cordova-plugin-code-push/issues/586 for detail.

**Comment by Krasavinigor** [comment]
Hi @bbialas!
Sorry for the delay.
Thank you for contributing. You've done good and useful job=)
I just leave one comment to code.

File: src/ios/CodePush.m [hunk 1]
```diff
@@ -401,10 +401,14 @@ - (void)loadURL:(NSURL*)url {
     // maintaining back-compat with Cordova iOS 3.9.0, we need to conditionally
     // use the WebViewEngine for performing navigations only if the host app
     // is running 4.0.0+, and fallback to directly using the WebView otherwise.
-#ifdef __CORDOVA_4_0_0
+#if WK_WEB_VIEW_ONLY
```
[the remark this answers is not in the document]
Comment (bbialas) [reply src/ios/CodePush.m:404]: Hello @Krasavinigor , thank you for the review. You are right the code duplication is there, but the question is, if this logical condition would be proper:
```
#if WK_WEB_VIEW_ONLY && defined(__CORDOVA_4_0_0)
    [self.webViewEngine loadRequest:[NSURLRequest requestWithURL:url]];
#else
    [(UIWebView*)self.webView loadRequest:[NSURLRequest requestWithURL:url]];
#endif
```
If we do it like that, then we exclude the usage of WKWebView when somebody uses the plugin without explicitly specifying the flag in config.xml and when his Cordova version supports it at the same time.
Maybe it should be more like:
```
#if (WK_WEB_VIEW_ONLY && defined(__CORDOVA_4_0_0)) || defined(__CORDOVA_4_0_0)
    [self.webViewEngine loadRequest:[NSURLRequest requestWithURL:url]];
#else
    [(UIWebView*)self.webView loadRequest:[NSURLRequest requestWithURL:url]];
#endif
```
What do you think?

File: README.md [hunk 2]
```diff
@@ -116,6 +116,16 @@ With the CodePush plugin installed, configure your app to use it via the followi
  
 You are now ready to use the plugin in the application code. See the [sample applications](/samples) for examples and the API documentation for more details.
 
+## iOS specific options
```
Comment (alexandergoncharov) [inline README.md:119]: I think that this info should be as part of the previous section. Not the different section. 
You can add this info like `Note`. For example https://github.com/microsoft/cordova-plugin-code-push/pull/588/files#diff-04c6e90faac2675aa89e2176d2eec7d8R155

File: src/ios/CodePush.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (Krasavinigor) [reply src/ios/CodePush.m:404]: @bbialas
Yes, you're right! It will be better. Thanks!

File: README.md (hunk 2 above)
Comment (bbialas) [reply README.md]: Changed as requested. Thanks for the review.

**Comment by bbialas** [comment]
All requested changes added.

**Comment by swbradsh** [comment]
For what its worth, I pulled in the changes made by @bbialas this afternoon, and my app was finally able to submit to Apple without the UIWebView warning.

**Comment by l3ender** [comment]
Thank you for merging/releasing this!

**Pull request merged**